keps/sig-cluster-lifecycle/kubeadm/4656-add-instance-configuration/README.md

```diff
@@ -0,0 +1,374 @@
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The proposal is to remove the Container Runtime Interface (CRI) socket annotation from the Node object in Kubernetes, which is currently added during the "init upload-config" phase in Kubeadm. This annotation is used to specify the CRI socket endpoint used by the kubelet on each node for communication with the container runtime. 
+
+Instead of relying on this annotation, the proposal recommends not relying on this annotation, but instead creating an instance config per node and overriding the `ContainerRuntimeEndpoint` in the global kubelet config when kubadm init and join. This will eliminate the need for kubeadm to store CRI socket configuration on each node, instead relying on the Kubernetes configuration file to specify this information.
+
+## Motivation
+
+Currently, kubeadm adds a CRI socket annotation to the Node object during the "init upload-config" phase, which specifies the endpoint for the CRI that is being used by the kubelet on each node. This annotation is persistent on the Node object, even if the kubelet is updated or the CRI is changed.
+
+After migrating the container runtime endpoint flag to the instance configuration, we can use it
+Set the cri socket in the global kubelet configuration by overriding the `ContainerRuntimeEndpoint` field in `kubelet-config.yml`.
+
+### Goals
+
+* Remove the use of CRI socket annotation on Node object.
+* We will prioritize respecting the local setting over the global one.
+* Add `kubelet-instace-config.yml` and use it to override the `ContainerRuntimeEndpoint` field in `kubelet-config.yml`.
+
+### Non-Goals
+
+Update the CRI socket annotation on Node object to be the latest.
```

Comment:
i don't think this is a non-goal.

you could add
- Continue maintaining CRI socket paths on Node objects